azurelinuxagent/common/protocol/wire.py

```diff
@@ -604,35 +606,39 @@ def hgap_download(uri):
 
         return self._download_with_fallback_channel(download_type, uris, direct_download=direct_download, hgap_download=hgap_download)
 
-    def download_extension(self, uris, destination, use_verify_header, on_downloaded=lambda: True):
```

Comment:
renamed this method since it can be used to download the agent as well; also, now this method can only be used to download ZIPs (we don't download any other type of files) and the expansion of the package is done here, instead of in the caller.